### PR TITLE
Fix Popover bug with fixed positioned children

### PR DESCRIPTION
**Pull request by dleroux**
### WHY are these changes introduced?

Because transformed elements create a containing block for all its positioned descendants, [a recent change](https://github.com/Shopify/polaris-react/pull/4580) to the Popover breaks `react-draggable-hoc`.

### WHAT is this pull request doing?

removes the transform on the popover-overlay when it's opened 

**Current:**
https://user-images.githubusercontent.com/1229901/142909448-a6880dc9-2d4c-48f7-ac3f-89602dd28cf1.mov

**Fixed:**
https://user-images.githubusercontent.com/1229901/142909436-fd697dd9-fe91-4453-9011-3a1b781962ef.mov

### How to 🎩

In storybook ensure this didn't break the animation on the popover.

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

File: src/components/Popover/Popover.scss [hunk 1]
```diff
@@ -27,7 +27,7 @@ $vertical-motion-offset: rem(-5px);
 
 .PopoverOverlay-open {
   opacity: 1;
-  transform: translateY(0);
+  transform: none;
```
Comment (alex-page) [inline src/components/Popover/Popover.scss:30]: Does this affect the motion? I haven't animated `none` versus `translateY(0)`.

**Review by danny-williams — APPROVED** [review]
The video you link @dleroux LGTM! 👍

File: src/components/Popover/Popover.scss (hunk 1 above)
Comment (dleroux) [reply src/components/Popover/Popover.scss:30]: When you trigger the popover, the class `entering` with `translateY(0)` is applied, which starts the transition. When the popover has opened, `none` is applied, which essentially removes the transform altogether. It never transitions to `none`.

Comment (alex-page) [reply src/components/Popover/Popover.scss:30]: TY. Legend.

**Comment by danny-williams** [comment]
@dleroux still seeing this issue in prod. Assuming we're held up by code freeze, ya?

**Pull request merged**